app/src/main/java/com/islaharper/dawnofandroid/presentation/screens/login/LoginViewModel.kt

```diff
@@ -0,0 +1,88 @@
+package com.islaharper.dawnofandroid.presentation.screens.login
+
+import androidx.compose.runtime.MutableState
+import androidx.compose.runtime.State
+import androidx.compose.runtime.mutableStateOf
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import com.islaharper.dawnofandroid.domain.model.ApiResponse
+import com.islaharper.dawnofandroid.domain.model.ApiTokenRequest
+import com.islaharper.dawnofandroid.domain.model.MessageBarState
+import com.islaharper.dawnofandroid.domain.useCases.readSignedInState.ReadSignedInStateUseCase
+import com.islaharper.dawnofandroid.domain.useCases.saveSignedInState.SaveSignedInStateUseCase
+import com.islaharper.dawnofandroid.domain.useCases.verifyToken.VerifyTokenUseCase
+import com.islaharper.dawnofandroid.util.Resource
+import dagger.hilt.android.lifecycle.HiltViewModel
+import kotlinx.coroutines.launch
+import java.net.ConnectException
+import java.net.SocketTimeoutException
+import javax.inject.Inject
+
+@HiltViewModel
+class LoginViewModel @Inject constructor(
```

Comment:
I think it would be good to add tests to this `LoginViewModel` as part of this PR so as to make sure we are designing it with testability in mind

Reply:
Committed changes for other comments. Busy with Unit tests